src/geojson/geojson.ml

```diff
@@ -265,24 +265,27 @@ module Make (J : Geojson_intf.Json) = struct
   end
 
   module Feature = struct
-    type t = Geometry.t option * json option
+    type id = [ `string of string | `int of int ]
 
-    let geometry = fst
-    let properties = snd
+    type t = {
+      id : id option;
+      geometry : Geometry.t option;
+      properties : json option;
+    }
 
     let of_json json =
       match J.find json [ "type" ] with
       | Some typ -> (
           match J.to_string typ with
           | Ok "Feature" -> (
               match
-                (J.find json [ "geometry" ], J.find json [ "properties" ])
+                ( J.find json [ "id" ],
```

Comment:
Note that the `of_json` function needs to return a `(t, [ `Msg of string ]) result` -- to help a little, I think this code can be simplified to something like:

```ocaml
match J.to_string typ with
  | Ok "Feature" -> (
      let id = J.find json [ "id" ] in
      let geometry = J.find json [ "geometry" ] in
      let properties = J.find json [ "properties" ] in
      let id = Option.map id_of_json id in 
      let geometry = Option.map (decode_or_err Geometry.of_json) geometry in 
      Ok { id; geometry; properties}
```

You'll need to implement `id_of_json`